clients/keycloak/src/main/kotlin/DefaultKeycloakClient.kt

```diff
@@ -383,6 +383,51 @@ class DefaultKeycloakClient(
         }.getOrElse {
             throw KeycloakClientException("Failed to load client roles for user '${id.value}'.", it)
         }.body()
+
+    override suspend fun addUserToGroup(username: UserName, groupName: GroupName) {
+        // FIXME: Unfortunately, no way to distinguish if there was a technical problem or the user was not found.
+        val user = try { getUser(username) } catch (e: KeycloakClientException) {
+            throw UserNotFoundException("User with username '${username.value}' not found.", e)
+        }
+        val group = getGroup(groupName)
+
+        runCatching {
+            httpClient.put("$apiUrl/users/${user.id.value}/groups/${group.id.value}")
+        }.onFailure {
+            throw KeycloakClientException(
+                "Failed to add user '{$username.value}' to group '{$groupName}'.",
+                it
+            )
+        }
+    }
+
+    override suspend fun removeUserFromGroup(username: UserName, groupName: GroupName) {
+        // FIXME: Unfortunately, no way to distinguish if there was a technical problem or the user was not found.
```

Comment:
Maybe the code for obtaining a user and a group by name could be extracted into helper methods. This will be needed for the other endpoints as well. If the methods returned a `Result`, they could be nicely integrated with further logic that is using `runCatching` (modulo Kotlin's limited supported for monads of course).